### ISSUE TITLE
Incomplete Documentation - man 5 systemd.unit - "Difference between PartOf and BindsTo in a systemd unit"

### ISSUE THREAD

**Issue by thx1111** [systemd/systemd#23194]
As of: systemd 250.4

Descriptions of the Unit File declarations, distinguishing "Requires=", "PartOf=", and "BindsTo=", are incomplete, and thereby, confusing, or confused.

1) Reference the investigative report at [Difference between PartOf and BindsTo in a systemd unit](https://pychao.com/2021/02/24/difference-between-partof-and-bindsto-in-a-systemd-unit/) .

2) Expand the Documentation at man 5 systemd.unit to include descriptions of the distinctions illuminated within the above report.

3) Emphasize more strongly, _within the descriptions of each of "Requires=", "PartOf=", and "BindsTo="_, that the *only* dependency declarations within these systemd unit declarations that will create a "Reverse Start" dependency between units is using "WantedBy=" or "RequiredBy=", *and* that the *only* way to specify a "WantedBy=" or "RequiredBy=" dependency is from the Unit File Install section, *and* that the *only* way to make this "WantedBy=" or "RequiredBy" dependency effective is to actually issue the command "systemd enable" referencing the Unit File holding the "WantedBy=" or "RequiredBy=" declaration.

It would also helpful to introduce the terms "forward dependency" and "reverse dependency" early in the dependency declaration descriptions, prior to their introduction in "Table 3", to allow the reader to more easily conceptualize and distinguish "the dependent Unit being acted upon" and the "the independent Unit which is initiating the action".  The "Reverse Start" dependency is very restricted in systemd and requires manual intervention.

Some suggested text:
Only "BindsTo=" provides a Reverse Abnormal-Stop dependency.
Only "PartOf=" does NOT provide a Forward Start dependency.
Otherwise "Requires=", "BindsTo=", and "PartOf=" behave in exactly the same way.
But note that Device Units are not made active by a Forward Start dependency, and would generally be invoked from a Template Service Unit File, so that a Forward Start dependency declaration involving a Device Unit would instead reference this corresponding Template Service Unit File, rather than the Device Unit itself.  However, a Device Unit *may* be referenced directly in a Reverse Start dependency declaration, using "WantedBy=" or "RequiredBy=".

"systemd" is like doorknobs - they are both easy to use, once you know their secrets, but will block your way without that knowledge.

**Comment by instantlinux** [comment]
In plain English: please tell me what directives I need in unit files A and B to ensure that whenever I invoke `systemd restart A.service`, the B service _also_ gets restarted?

**Comment by instantlinux** [comment]
In plain English: please tell me what directives I need in unit files A and B to ensure that whenever I invoke `systemctl restart A.service`, the B service also gets restarted?

**Comment by copolycube** [comment]
@instantlinux  if it helps, I managed to do it by adding this in the `[Unit]` section of `B.service`:

```
[Unit]
...
After=A.target
Wants=A.target
PartOf=A.target
...
```
(no change to `A.service` definition)

**Comment by alacham** [comment]
@copolycube that indded works, but I was not able to replicate the same behaviour when instead of calling restart I called `stop A.service` and then `start A.service`. To achieve that behaviour I needed to add an `ExecStartPost` action to `A.service` which IMHO isn't perfect.
